src/v/transform/logging/rpc_client.cc

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright 2024 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#include "rpc_client.h"
+
+#include "hashing/murmur.h"
+#include "model/namespace.h"
+#include "transform/logging/logger.h"
+#include "transform/logging/record_batcher.h"
+
+#include <seastar/core/future.hh>
+#include <seastar/coroutine/as_future.hh>
+
+namespace transform::logging {
+
+rpc_client::rpc_client(
+  rpc::client* rpc_client, cluster::metadata_cache* metadata_cache)
+  : _rpc_client(rpc_client)
+  , _metadata_cache(metadata_cache) {}
+
+ss::future<errc>
+rpc_client::write(model::partition_id pid, io::json_batches batches) {
+    model::topic_namespace_view ns_tp{model::transform_log_internal_nt};
+    auto cfg = _metadata_cache->get_topic_cfg(ns_tp);
+    if (!cfg.has_value()) {
+        co_return errc::topic_not_found;
+    }
+
+    auto max_batch_size = cfg->properties.batch_max_bytes.value_or(
+      config::shard_local_cfg().kafka_batch_max_bytes());
+
+    record_batcher batcher{max_batch_size};
+
+    while (!batches.empty()) {
+        auto json_batch = std::move(batches.front());
+        batches.pop_front();
+        while (!json_batch.events.empty()) {
+            auto ev = std::move(json_batch.events.front());
+            json_batch.events.pop_front();
+            batcher.append(json_batch.name.copy(), std::move(ev));
+        }
+    }
```

Comment:
Are we worried about stalls here at all?

Reply:
not sure, but iiuc adding a `maybe_wait` doesn't hurt anything. good call

Reply:
done